src/main/java/com/ericsson/ei/frontend/BackEndInformationController.java

```diff
@@ -16,35 +16,107 @@
+        try {
+            String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            backEndInstancesUtils.setInstances(new JsonParser().parse(body).getAsJsonArray());
+            backEndInstancesUtils.writeIntoFile();
+            for (BackEndInformation backEndInformation : backEndInstancesUtils.getInformation()) {
+                if (backEndInformation.isActive()) {
+                    backEndInstancesUtils.setBackEndProperties(backEndInformation);
+                }
+            }
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
-    @CrossOrigin
-    @RequestMapping(value = "/switch-backendByMainPage", method = RequestMethod.POST)
-    ResponseEntity<String> switchBackEndInstanceByMainPage(Model model, HttpServletRequest request);
-
-    @CrossOrigin
     @RequestMapping(value = "/switch-backend", method = RequestMethod.DELETE)
-    ResponseEntity<String> deleteBackEndInstance(Model model, HttpServletRequest request);
+    public ResponseEntity<String> deleteBackEndInstance(Model model, HttpServletRequest request) {
+        try {
+            String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            backEndInstancesUtils.setInstances(new JsonParser().parse(body).getAsJsonArray());
+            backEndInstancesUtils.writeIntoFile();
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
-    @CrossOrigin
     @RequestMapping(value = "/add-instances", method = RequestMethod.POST)
-    ResponseEntity<String> addInstanceInformation(Model model, HttpServletRequest request);
+    public ResponseEntity<String> addInstanceInformation(Model model, HttpServletRequest request) {
+        try {
+            String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            JsonObject instance = new JsonParser().parse(body).getAsJsonObject();
+            if (!backEndInstancesUtils.checkIfInstanceAlreadyExist(instance)) {
+                backEndInstancesUtils.getInstances().add(instance);
+                backEndInstancesUtils.writeIntoFile();
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>("Instance already exist", HttpStatus.BAD_REQUEST);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value = "/switchBackend", method = RequestMethod.POST)
+    public ResponseEntity<String> switchBackEndInstanceByMainPage(Model model, HttpServletRequest request) {
```

Comment:
What do you mean "ByMainPage"? ,, switch is only steering to which address to do RestApi calls to.

Reply:
"frontendServiceUrl" and same variable in Bridge should be updated with new url when switch is made.

Reply:
We have two chooses in switching backends first from drop-list(current) and second from page